common/cluster-api/src/main/proto/queue.proto

```diff
@@ -180,6 +180,10 @@ message ValidateDeviceX509CertRequestMsg {
   string hash = 1;
 }
 
+message ValidateOrCreateDeviceX509CertRequestMsg {
```

Comment:
Could we just add a field to ProvisionDeviceCredentialsMsg?

Reply:
I don't think so. This is not a classic provision strategy, in most cases, it will work as before - connect to devices, so I believe that sending CredentialsDataProto with empty fields to connect is a bad approach.